libs/variant/include/variant/variant.hpp

```diff
@@ -576,4 +584,167 @@ void Variant::IterateObject(Function const &function) const
+    serializer << typecode;
+    switch (var.type())
+    {
+    case Type::Type::UNDEFINED:
+    {
+      return;
+    }
+
+    case Type::Type::NULL_VALUE:
+    {
+      return;
+    }
+    case Type::Type::INTEGER:
+    {
+      serializer << var.As<int>();
+      return;
+    }
+    case Type::Type::FLOATING_POINT:
+    {
+      serializer << var.As<double>();
+      return;
+    }
+    case Type::Type::FIXED_POINT:
+    {
+      serializer << var.As<int>();
+      return;
+    }
+    case Type::Type::BOOLEAN:
+    {
+      serializer << var.As<bool>();
+      return;
+    }
+    case Type::Type::STRING:
+    {
+      serializer << var.As<std::string>();
+      return;
+    }
+    case Type::Type::ARRAY:
+    {
+      auto sz = static_cast<uint32_t>(var.size());
+      serializer << sz;
+      for(std::size_t i=0;i<var.size();i++)
+      {
+        Serialize(serializer, var[i]);
+      }
+      return;
+    }
+    case Type::Type::OBJECT:
+    {
+      auto sz = static_cast<uint32_t>(var.size());
+      serializer << sz;
+      var.IterateObject([&](auto const &k, auto const &v){
+          serializer << k;
+          Serialize(serializer, v);
+          return true;
+        });
+      return;
+    }
+    };
+    throw std::runtime_error{"Variant has unknown type."};
+  }
+
+  template <typename Deserializer>
+  static void Deserialize(Deserializer &deserializer, Type &var)
+  {
+    int typecode;
+    deserializer >> typecode;
+    auto type = static_cast<Type::Type>(typecode);
+    switch (type)
+    {
+    case Type::Type::UNDEFINED:
+      {
+        var = fetch::variant::Variant::Undefined();
+        return;
+      }
+    case Type::Type::NULL_VALUE:
+      {
+        var = fetch::variant::Variant::Null();
+        return;
+      }
+    case Type::Type::INTEGER:
+      {
+        int64_t tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::FLOATING_POINT:
+      {
+        double tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::FIXED_POINT:
+      {
+        int64_t tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::BOOLEAN:
+      {
+        bool tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::STRING:
+      {
+        std::string tmp;
+        deserializer >> tmp;
+        var = tmp;
+        return;
+      }
+    case Type::Type::ARRAY:
+    {
+      uint64_t    count;
+      deserializer >> count;
+      var = fetch::variant::Variant::Array(count);
+      for (std::size_t i = 0; i < count; i++)
+      {
+        Type v;
+        Deserialize(deserializer, v);
+        var[i]=v;
+      }
+      return;
+    }
+    case Type::Type::OBJECT:
+    {
+      uint64_t    count;
+      deserializer >> count;
+      var = fetch::variant::Variant::Object();
+      for (uint64_t i = 0; i < count; i++)
```

Comment:
Just comment: uint64_t is alright, but std::size_t is standard type used for indexing, and it is also type used in `Variant::operator[](std::size_t)`

Reply:
It's mostly for the type match with the deserialised counter -- because there's no ser/des for size_t